lib/std/process/Child.zig

```diff
@@ -1444,6 +1435,33 @@ test windowsCreateProcessSupportsExtension {
     try std.testing.expect(windowsCreateProcessSupportsExtension(&[_]u16{ '.', 'e', 'X', 'e', 'c' }) == null);
 }
 
+test "spawnRelative" {
+    const root = if (builtin.target.os.tag == .windows) "\\" else "/";
+
+    var args = std.ArrayList([]const u8).init(std.testing.allocator);
+    defer args.deinit();
+
+    const fake_exe_name = try std.fmt.allocPrint(std.testing.allocator, "{x}", .{
+        std.crypto.random.int(u64),
+    });
+    defer std.testing.allocator.free(fake_exe_name);
+
+    const fake_exe_path = try std.fs.path.join(std.testing.allocator, &[_][]const u8{ root, "..", "..", "..", fake_exe_name });
```

Comment:
Note that this will create an absolute path. On windows it will be a "rooted" path (relative to the root of the CWD's drive), e.g. `\..\..\..\<random int>`. Is that intentional? It doesn't seem to match your description of the problem.